OSDOCS#5384: Document the explicit list of required credential permissions for Azure (#56274)
openshift/openshift-docs: modules/minimum-required-permissions-upi-azure.adoc

> // Module included in the following assemblies:

Author: Addressed

// Module included in the following assemblies:
//
// * installing/installing_azure/installing-azure-user-infra.adoc

[id="minimum-required-azure-permissions_{context}"]
= Required Azure permissions in user-provisioned infrastructure

> When you assign the contributor and user access administrator roles to the service principal, you automatically grant all the required permissions. To deploy and destroy an {product-title} cluster with installer-provisioned infrastructure, the service principal requires the following permissions.

Reviewer: Similar to IPI

Author: Addressed

Following are the permissions required for creating the {product-title} cluster.

.Required permissions for creating authorization resources
[%collapsible]
====
* `Microsoft.Authorization/policies/audit/action`
* `Microsoft.Authorization/policies/auditIfNotExists/action`
* `Microsoft.Authorization/roleAssignments/read`
* `Microsoft.Authorization/roleAssignments/write`
====

> .Required permissions for creating compute resources

Reviewer: 3 permissions should be moved here from optional, as suggested in other comments.

* `Microsoft.Network/networkSecurityGroups/read`
* `Microsoft.Network/networkSecurityGroups/securityRules/delete`
* `Microsoft.Network/networkSecurityGroups/securityRules/read`
* `Microsoft.Network/networkSecurityGroups/securityRules/write`
* `Microsoft.Network/networkSecurityGroups/write`
* `Microsoft.Network/privateDnsZones/A/read`
* `Microsoft.Network/privateDnsZones/A/write`
* `Microsoft.Network/privateDnsZones/A/delete`
* `Microsoft.Network/privateDnsZones/SOA/read`
* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/write`
* `Microsoft.Network/privateDnsZones/write`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/publicIPAddresses/join/action`
* `Microsoft.Network/publicIPAddresses/read`
* `Microsoft.Network/publicIPAddresses/write`
* `Microsoft.Network/virtualNetworks/join/action`
* `Microsoft.Network/virtualNetworks/read`
* `Microsoft.Network/virtualNetworks/subnets/join/action`
* `Microsoft.Network/virtualNetworks/subnets/read`
* `Microsoft.Network/virtualNetworks/subnets/write`
* `Microsoft.Network/virtualNetworks/write`
====

> .Required permissions for checking health resources

Reviewer: Similar to IPI

Author: Addressed

[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/InProgress/action`
* `Microsoft.Resourcehealth/healthevent/Pending/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

.Required permissions for creating subscription resources
[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourceGroups/read`
* `Microsoft.Resources/subscriptions/resourcegroups/write`
* `Microsoft.Resources/tags/write`

> ====

Reviewer: Similar to IPI. can be separated into 2 parts.

.Required permissions for creating storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/blobServices/read`
* `Microsoft.Storage/storageAccounts/blobServices/containers/write`
* `Microsoft.Storage/storageAccounts/fileServices/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/write`
* `Microsoft.Storage/storageAccounts/fileServices/shares/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
* `Microsoft.Storage/storageAccounts/read`
* `Microsoft.Storage/storageAccounts/write`
====

.Required permissions for creating deployment
[%collapsible]
====
Microsoft.Resources/deployments/read
Microsoft.Resources/deployments/write
Microsoft.Resources/deployments/validate/action
Microsoft.Resources/deployments/operationstatuses/read
====

> Following are the permissions required for deleting the cluster.

Reviewer: Similar to IPI

* `Microsoft.Network/privateDnsZones/A/read`
* `Microsoft.Network/privateDnsZones/delete`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/virtualNetworks/delete`
====

.Required permissions for checking health resources
[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

.Required permissions for deleting resource groups
[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourcegroups/delete`
====

.Required permissions for deleting storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
====

.Optional permissions for creating marketplace virtual machine resources
[%collapsible]
====
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/read`
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/write`
====

.Optional permissions for creating compute resources
[%collapsible]
====
* `Microsoft.Compute/images/read`

> * `Microsoft.Compute/images/write`

Author: Addressed

> * `Microsoft.Compute/images/delete`

Author: Addressed

* `Microsoft.Compute/diskEncryptionSets/read`
* `Microsoft.Compute/diskEncryptionSets/write`
* `Microsoft.Compute/diskEncryptionSets/delete`

> * `Microsoft.Compute/virtualMachines/deallocate/action`

Reviewer: `Microsoft.Compute/virtualMachines/deallocate/action` is required permission for UPI. Please move it to required section of creation

Only 3 `diskEncryptionSets` related permissions would be optional. Rest are required for UPI.

Author: Addressed

====

.Optional resources for creating key vault resources
[%collapsible]
====
* `Microsoft.KeyVault/vaults/read`
* `Microsoft.KeyVault/vaults/write`
* `Microsoft.KeyVault/vaults/delete`
* `Microsoft.KeyVault/vaults/deploy/action`
* `Microsoft.KeyVault/vaults/keys/read`
* `Microsoft.KeyVault/vaults/keys/write`

> ====

Author: Addressed

[NOTE]
====

> You can create a custom role with all the permissions and assign that custom role to the service principal. To create a custom role in the Azure portal, see the Azure custom roles in the Azure documentation.

Reviewer: Similar to IPI

Author: Addressed

* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/A/read`
* `Microsoft.Network/privateDnsZones/delete`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/virtualNetworks/delete`
====

.Required permissions for checking health of resources
[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

.Required permissions for deleting resource groups
[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourcegroups/delete`
====

.Required permissions for deleting storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
====

[NOTE]
====
You can create a custom role with all the permissions and assign that custom role to the service principal. To create a custom role in the Azure portal, see link:https://learn.microsoft.com/en-us/azure/role-based-access-control/custom-roles[Azure custom roles] in the Azure documentation.

> ====

Reviewer: @jinyunma As discussed, we will add a new NOTE here for UPI part. 

```
- For cluster installation, the permissions related to resource group creation should be scoped to subscription level, and later on further permissions can be scoped to created resource group level. If the Public DNS zone is present in different resource group, then Network DNS zone related permissions should always be subscription level.

- For cluster deletion, all the permissions can be scoped to subscription level.

```

Reviewer: We will add a new note just after the existing note.

Reviewer: For {product-title} cluster installation, you must scope the permissions related to resource group creation, to a subscription level. After resource group is created, you can scope the rest of the permissions to the created resource group level. If the public DNS zone is present in a different resource group, then the network DNS zone related permissions must always be at subscription level.

For deleting the {product-title} cluster, you can scope all the permissions to a subscription level.

// Module included in the following assemblies:
//
// * installing/installing_azure/installing-azure-user-infra.adoc

> [id="minimum-required-azure-permissions_{context}"]

Reviewer 2: This ID is the same as the one in modules/minimum-required-permissions-ipi-azure.adoc. Please change both module IDs so that they match the corresponding filename. This one should be: `[id="minimum-required-permissions-upi-azure_{context}"]`

Thanks!

// Module included in the following assemblies:
//
// * installing/installing_azure/installing-azure-user-infra.adoc

[id="minimum-required-azure-permissions_{context}"]
= Required Azure permissions in user-provisioned infrastructure

> When you assign the contributor and user access administrator roles to the service principal, you automatically grant all the required permissions. To deploy and destroy an {product-title} cluster with user-provisioned infrastructure, the service principal requires the following permissions.

Reviewer: Similar highlight should be applied here also, for name of the roles. Thanks!

Author: Addressed

* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

.Required permissions for creating resource groups
[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourceGroups/read`
* `Microsoft.Resources/subscriptions/resourcegroups/write`
====

.Required permissions for creating resource tags
[%collapsible]
====
* `Microsoft.Resources/tags/write`
====

.Required permissions for creating storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/blobServices/read`
* `Microsoft.Storage/storageAccounts/blobServices/containers/write`
* `Microsoft.Storage/storageAccounts/fileServices/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/write`
* `Microsoft.Storage/storageAccounts/fileServices/shares/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
* `Microsoft.Storage/storageAccounts/read`
* `Microsoft.Storage/storageAccounts/write`
====

.Required permissions for creating deployment
[%collapsible]
====
Microsoft.Resources/deployments/read
Microsoft.Resources/deployments/write
Microsoft.Resources/deployments/validate/action

> Microsoft.Resources/deployments/operationstatuses/read

Reviewer: Bullet points and `` are missing

Author: Addressed

* `Microsoft.Storage/storageAccounts/read`
* `Microsoft.Storage/storageAccounts/write`
====

.Required permissions for creating deployment
[%collapsible]
====
* `Microsoft.Resources/deployments/read`
* `Microsoft.Resources/deployments/write`
* `Microsoft.Resources/deployments/validate/action`
* `Microsoft.Resources/deployments/operationstatuses/read`
====

.Optional permissions for creating marketplace virtual machine resources
[%collapsible]
====
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/read`
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/write`
====

.Optional permissions for creating compute resources
[%collapsible]
====
* `Microsoft.Compute/diskEncryptionSets/read`
* `Microsoft.Compute/diskEncryptionSets/write`
* `Microsoft.Compute/diskEncryptionSets/delete`
====

.Optional resources for creating key vault resources
[%collapsible]
====
* `Microsoft.KeyVault/vaults/read`
* `Microsoft.KeyVault/vaults/write`
* `Microsoft.KeyVault/vaults/delete`
* `Microsoft.KeyVault/vaults/deploy/action`
* `Microsoft.KeyVault/vaults/keys/read`
* `Microsoft.KeyVault/vaults/keys/write`

> ====

Reviewer: Please merge them as done in IPI part with : "Optional permissions for enabling user-defined disk encryption set"

And as discussed in [CORS-2559](https://issues.redhat.com/browse/CORS-2559), please add `Microsoft.Features/providers/features/register/action` here.

Author: Addressed

// Module included in the following assemblies:
//
// * installing/installing_azure/installing-azure-user-infra.adoc

[id="minimum-required-azure-permissions_{context}"]
= Required Azure permissions for user-provisioned infrastructure

When you assign `Contributor` and `User Access Administrator` roles to the service principal, you automatically grant all the required permissions.

> Following are the permissions required for creating the {product-title} cluster on Azure public cloud.

Reviewer 3: ```suggestion
The following permissions are required for creating an {product-title} cluster on Microsoft Azure.
```
Rewording suggestion

Author: Addressed

* `Microsoft.Network/privateDnsZones/A/delete`
* `Microsoft.Network/privateDnsZones/SOA/read`
* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/write`
* `Microsoft.Network/privateDnsZones/write`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/publicIPAddresses/join/action`
* `Microsoft.Network/publicIPAddresses/read`
* `Microsoft.Network/publicIPAddresses/write`
* `Microsoft.Network/virtualNetworks/join/action`
* `Microsoft.Network/virtualNetworks/read`
* `Microsoft.Network/virtualNetworks/subnets/join/action`
* `Microsoft.Network/virtualNetworks/subnets/read`
* `Microsoft.Network/virtualNetworks/subnets/write`
* `Microsoft.Network/virtualNetworks/write`
====

.Required permissions for checking health of resources
[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/InProgress/action`
* `Microsoft.Resourcehealth/healthevent/Pending/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

> .Required permissions for creating resource group

Reviewer 3: ```suggestion
.Required permissions for creating resource groups
```

Author: @bscott-rh Here the singular form of `group` was used intentionally because we can create only a single resource group in the cluster. cc @chiragkyal

Author: But I will double check with @chiragkyal if resource groups can be used

Reviewer: AFAIK, we only need one resource group to deploy an OCP cluster. Hence I suggested to use `group` in place of `groups`. cc @jinyunma

Reviewer 4: For UPI, it's true. 
But for IPI, if user deploys on existing vNet, then the vNet related resources can be allocated in another resource group.

Reviewer: @jinyunma, even if the user deploys on existing vNet, is there any possibility that the installer will handle (create/delete) two resource groups? I am not sure about that.

Reviewer 4: no, installer is only responsible for destroying the resource group where installer created, there is only one resource group installer handling.

Reviewer 3: Sounds good. In that case, it should be `for creating a resource group`.

Reviewer 3: @xenolinux looks like this still needs to be updated to `a resource group`.

[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourceGroups/read`
* `Microsoft.Resources/subscriptions/resourcegroups/write`
====

.Required permissions for creating resource tags
[%collapsible]
====
* `Microsoft.Resources/tags/write`
====

.Required permissions for creating storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/blobServices/read`
* `Microsoft.Storage/storageAccounts/blobServices/containers/write`
* `Microsoft.Storage/storageAccounts/fileServices/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/read`
* `Microsoft.Storage/storageAccounts/fileServices/shares/write`
* `Microsoft.Storage/storageAccounts/fileServices/shares/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
* `Microsoft.Storage/storageAccounts/read`
* `Microsoft.Storage/storageAccounts/write`
====

> .Required permissions for creating deployment

Reviewer 3: ```suggestion
.Required permissions for creating deployments
```

Author: Addressed

[%collapsible]
====
* `Microsoft.Resources/deployments/read`
* `Microsoft.Resources/deployments/write`
* `Microsoft.Resources/deployments/validate/action`
* `Microsoft.Resources/deployments/operationstatuses/read`
====

.Optional permissions for creating marketplace virtual machine resources
[%collapsible]
====
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/read`
* `Microsoft.MarketplaceOrdering/offertypes/publishers/offers/plans/agreements/write`
====

> .Optional permissions for enabling user-defined disk encryption set

Reviewer 4: I searched in UPI doc, it doesn't mention to enable disk encryption, so I think we can remove this part from UPI doc. When doc describes this part here, then add them back.

@chiragkyal wdyt?

Reviewer 4: Double checked with @sunzhaohua2  from cloud team, the day-2 operation on machine-api (including disk encryption set) is same on both ipi and upi, no difference. @sunzhaohua2 please correct me if I 'm wrong.

So I'm okay to keep this part as day2 operation.

@xenolinux if no more comments here, same update should be here as IPI to reword `Optional permissions for enabling user-defined disk encryption set`.

Reviewer: Let's be consistent throughout the docs. We haven't called day 1 and day 2 terms. So, I think `Optional permissions for enabling user-defined disk encryption set` would make much more sense.

[%collapsible]
====
* `Microsoft.Compute/diskEncryptionSets/read`
* `Microsoft.Compute/diskEncryptionSets/write`
* `Microsoft.Compute/diskEncryptionSets/delete`
* `Microsoft.KeyVault/vaults/read`
* `Microsoft.KeyVault/vaults/write`
* `Microsoft.KeyVault/vaults/delete`
* `Microsoft.KeyVault/vaults/deploy/action`
* `Microsoft.KeyVault/vaults/keys/read`
* `Microsoft.KeyVault/vaults/keys/write`
* `Microsoft.Features/providers/features/register/action`
====

> Following are the permissions required for deleting the {product-title} cluster in Azure public cloud.

Reviewer 3: ```suggestion
The following permissions are required for deleting an {product-title} cluster on Microsoft Azure.
```

Author: Addressed

.Required permissions for deleting compute resources
[%collapsible]
====
* `Microsoft.Compute/disks/delete`
* `Microsoft.Compute/galleries/delete`
* `Microsoft.Compute/galleries/images/delete`
* `Microsoft.Compute/galleries/images/versions/delete`
* `Microsoft.Compute/virtualMachines/delete`
* `Microsoft.Compute/images/delete`
====

.Required permissions for deleting identity management resources
[%collapsible]
====
* `Microsoft.ManagedIdentity/userAssignedIdentities/delete`
====

.Required permissions for deleting network resources
[%collapsible]
====
* `Microsoft.Network/dnszones/read`
* `Microsoft.Network/dnsZones/A/read`
* `Microsoft.Network/dnsZones/A/delete`
* `Microsoft.Network/dnsZones/CNAME/read`
* `Microsoft.Network/dnsZones/CNAME/delete`
* `Microsoft.Network/loadBalancers/delete`
* `Microsoft.Network/networkInterfaces/delete`
* `Microsoft.Network/networkSecurityGroups/delete`
* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/A/read`
* `Microsoft.Network/privateDnsZones/delete`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/virtualNetworks/delete`
====

> .Required permissions for checking health of resources

Reviewer 3: ```suggestion
.Required permissions for checking the health of resources
```

Author: Addressed

[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

> .Required permissions for deleting resource group

Reviewer 3: ```suggestion
.Required permissions for deleting resource groups
```

[%collapsible]
====
* `Microsoft.Resources/subscriptions/resourcegroups/delete`
====

.Required permissions for deleting storage resources
[%collapsible]
====
* `Microsoft.Storage/storageAccounts/delete`
* `Microsoft.Storage/storageAccounts/listKeys/action`
====

[NOTE]
====
You can create a custom role with all the permissions and assign that custom role to the service principal. To create a custom role in the Azure portal, see link:https://learn.microsoft.com/en-us/azure/role-based-access-control/custom-roles[Azure custom roles] in the Azure documentation.

> ====

Reviewer 3: I think this note is not needed since you are explaining it at the start of the module.

Author: Addressed

[NOTE]
====

> For installing the {product-title} cluster, you must scope the permissions related to resource group creation, to a subscription level. After resource group is created, you can scope the rest of the permissions to the created resource group level. If the public DNS zone is present in a different resource group, then the network DNS zone related permissions must always be at subscription level.

Reviewer 3: ```suggestion
To install {product-title} on Azure, you must apply the permissions to your subscription. Later, you can re-apply these permissions to the installer created resource group. If the public DNS zone is present in a different resource group, then the network DNS zone related permissions must always be applied to your subscription.
```
Rewording suggestion same as above

Author: Addressed

> For deleting the {product-title} cluster, you can scope all the permissions to a subscription level.

Reviewer 3: ```suggestion
You can apply all the permissions to your subscription when deleting an {product-title} cluster.
```
Rewording suggestion

Author: Addressed

Reviewer: For all the comments above, I am still wondering if `apply` will sound correct in place of `scope`. Technically `scope` is the appropriate term to use with permission.  @jinyunma wdyt ?

Reviewer: Sounds good to me; we need to make similar changes to all the similar places then. cc @xenolinux

Author: Yes. I will make the changes. I was wondering if we can keep the sentence like this - 
`You can scope all the permissions to your subscription when deleting an {product-title} cluster.` Does this sound good?

Author: Addressed

* `Microsoft.Network/privateDnsZones/A/delete`
* `Microsoft.Network/privateDnsZones/SOA/read`
* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/read`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/write`
* `Microsoft.Network/privateDnsZones/write`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/publicIPAddresses/join/action`
* `Microsoft.Network/publicIPAddresses/read`
* `Microsoft.Network/publicIPAddresses/write`
* `Microsoft.Network/virtualNetworks/join/action`
* `Microsoft.Network/virtualNetworks/read`
* `Microsoft.Network/virtualNetworks/subnets/join/action`
* `Microsoft.Network/virtualNetworks/subnets/read`
* `Microsoft.Network/virtualNetworks/subnets/write`
* `Microsoft.Network/virtualNetworks/write`
====

.Required permissions for checking health of resources
[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/InProgress/action`
* `Microsoft.Resourcehealth/healthevent/Pending/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

> .Required permissions for creating resource groups

Author: Addressed

* `Microsoft.Compute/virtualMachines/delete`
* `Microsoft.Compute/images/delete`
====

.Required permissions for deleting identity management resources
[%collapsible]
====
* `Microsoft.ManagedIdentity/userAssignedIdentities/delete`
====

.Required permissions for deleting network resources
[%collapsible]
====
* `Microsoft.Network/dnszones/read`
* `Microsoft.Network/dnsZones/A/read`
* `Microsoft.Network/dnsZones/A/delete`
* `Microsoft.Network/dnsZones/CNAME/read`
* `Microsoft.Network/dnsZones/CNAME/delete`
* `Microsoft.Network/loadBalancers/delete`
* `Microsoft.Network/networkInterfaces/delete`
* `Microsoft.Network/networkSecurityGroups/delete`
* `Microsoft.Network/privateDnsZones/read`
* `Microsoft.Network/privateDnsZones/A/read`
* `Microsoft.Network/privateDnsZones/delete`
* `Microsoft.Network/privateDnsZones/virtualNetworkLinks/delete`
* `Microsoft.Network/publicIPAddresses/delete`
* `Microsoft.Network/virtualNetworks/delete`
====

.Required permissions for checking health of resources
[%collapsible]
====
* `Microsoft.Resourcehealth/healthevent/Activated/action`
* `Microsoft.Resourcehealth/healthevent/Resolved/action`
* `Microsoft.Resourcehealth/healthevent/Updated/action`
====

> .Required permissions for deleting resource groups

Author: Addressed
